A19-tls-credential-API

```diff
@@ -0,0 +1,281 @@
+Title
+----
+* Author(s): yihuazhang
+* Approver: vjpai, markdroth
+* Status: Draft
+* Implemented in:
+* Last updated: September 3, 2018
+* Discussion at: https://groups.google.com/forum/#!topic/grpc-io/gMrROsizrtc
+
+## Abstract
+This proposal aims to incorporate SPIFFE-based mutual TLS into gRPC C core auth stack. Towards the goal, it proposes a new credential API model that 1) allows flexible addition of new TLS-related features (e.g., credential reload) without breaking the API and 2) supports both synchronous and asynchronous implementations of those features. It then proposes SPIFFE and HTTPS-based TLS credential API's that comply with the new model.
+
+## Background
+
+The current TLS implementation in gRPC C core has the following problems when it comes to support SPIFFE-based mutual TLS.
+
+1: It is originally designed to comply with HTTPS semantics which makes it difficult to be configured to support SPIFFE-based mutual TLS. For instance, the server authorization check in the current implementation is enforced locally by leveraging whitelisted endpoint information embedded in peer certificates while in SPIFFE use case, it may be enforced via an external server authorization check service.
```

Comment:
Please clarify what is server authorization. such as "_Server Authorization provides client-side checks that the servers clients connect to are legitimate._"